src/v/pandaproxy/schema_registry/exceptions.h

```diff
@@ -11,36 +11,37 @@
 
 #pragma once
 
-#include "pandaproxy/schema_registry/error.h"
-
-#include <fmt/format.h>
-
-#include <stdexcept>
-#include <string>
 #include <system_error>
 
 namespace pandaproxy::schema_registry {
 
-struct exception_base : std::exception {
-    explicit exception_base(std::error_condition err)
+// Similar to std::system_error but with more control over the message
+class exception_base : public std::exception {
+public:
+    explicit exception_base(std::error_code ec)
       : std::exception{}
-      , error{err}
-      , msg{err.message()} {}
-    exception_base(std::error_condition err, std::string_view msg)
+      , _ec{ec}
+      , _msg{ec.message()} {}
+    exception_base(std::error_code ec, std::string msg)
```

Comment:
why not sstring?

Reply:
I was torn. It's right at the boundary of interop with std, so I was trying to be compatible with https://en.cppreference.com/w/cpp/error/error_code/message, but also noexcept.